FLEDGE.md

```diff
@@ -626,6 +639,16 @@ The `BidFor240Minutes` interest group will have a positive priority if it was jo
 
 The `FilterOnDataFromServer` interest group will result in fetching `https://buyer1.com/bidder_signals?publisher=<...>&interest_groups=FilterOnDataFromServer,<...>`, and then if that result has a `perInterestGroupData.FilterOnDataFromServer.priorityVector` object, then that is used just like the `priorityVector` field from the other two examples, except that it's only used for filtering, not to set the priority (unless the group has a true `enableBiddingSignalsPrioritization` field).  A [user defined function](https://github.com/privacysandbox/fledge-docs/blob/main/key_value_service_trust_model.md#support-for-user-defined-functions-udfs) could be used on the FLEDGE Key-Value server to calculate that `priorityVector` value, and hence to decide if `FilterOnDataFromServer`'s `generateBid()` method is invoked or if it's filtered out.
 
+### 3.6 Currency Checking
+
+If participants in the auction need to deal with multiple currencies, they can optionally take advantage of automated currency checking. All of it operates on currency tags, which are required to contain 3 upper-case ASCII letters.
+
+If the `generateBid()` method returns a `bidCurrency`, and the `perBuyerCurrencies` for it is specified, their consistency will be checked, and if there is a mismatch, the bid will be dropped.  Both the configuration and the returned tag must be present for checking to take place.  The returned `bidCurrency` will be passed to `scoreAd()`'s `browserSignals.bidCurrency`, with unspecified currency rendered as `'???'`.
+
+Checking inside `scoreAd()` happens only inside component auctions.  If the auction modifies the bid, the modified version will be checked; if not, the passed-through bid from the original buyer will be. In either case, the currency will be checked both against the component auction's `sellerCurrency` and top-level auction's `perBuyerCurrencies` as applied to the component auction.  As before, both the bid and the configuration in question must be specified for the checking to take place. If there is a mismatch, the bid will not take part in the component auction.
```

Comment:
```suggestion
Additional currency checking after `scoreAd()` happens only inside component auctions.  If `scoreAd()` modifies the bid currency (i.e. returns a `bidCurrency`), the modified currency will be checked; if not, the passed-through bid currency from the original buyer will be. In either case, the currency will be checked both against the component auction's `sellerCurrency` and top-level auction's `perBuyerCurrencies` as applied to the component auction.  As before, both the bid and the configuration in question must be specified for the checking to take place. If there is a mismatch, the bid will not take part in the component auction.
```
I'm not sure my rewrite is correct.  Does "top-level auction's `perBuyerCurrencies` as applied to the component auction" mean that it applies to the buyer or component seller?

Reply:
It's not, but it's helpful, made some edits.  It's a question of whether scoreAd provides a bid, whether with bidCurrency or not.   As for the second question:t he component seller.

Conceptually, the component auction is a bidder in the top-level auction. Sometimes it forwards the buyer's original bid (including the currency tag, if any). Sometimes it provides its own bid (either including or omitting a currency tag).  That gets checked. 

(Though the perBuyerCurrencies thing is a bit different from how we behave elsewhere, so maybe a bit strange).

(If it provides a bid but not bidCurrency the check will always *pass*, but conceptually it happened).